Wireguard does not connect when using LAN8720 wired ethernet

I have been testing this component with WiFi, works perfectly. However, when switching to LAN8720 ethernet module, wireguard does not connect. I get the following repeated error message every 10s:

```
[D][wireguard:080]: WireGuard remote peer is offline (latest handshake timestamp not available)
```

The ESP32 is connected to the Internet and is reachable by Home Assistant, etc.

Relevant part of the config:

```yaml

esp32:
  board: esp32dev
  framework:
    type: arduino


external_components:
  - source: github://droscy/esphome@wireguard/main
    components:
      - wireguard
    refresh: 24h

ethernet:
  type: LAN8720
  mdc_pin: GPIO23
  mdio_pin: GPIO18
  clk_mode: GPIO17_OUT
  phy_addr: 0

wireguard:
  address: 10.80.0.3
  private_key: <edited>
  peer_public_key: <edited>
  peer_endpoint: "some.host.com"
  peer_port: 29432
  peer_persistent_keepalive: 25
  reboot_timeout: 0s
```

Hello @jpeletier, great work, many thanks. 

I still had a missing initialization for `err`

```
.piolibdeps/esp07/esp_wireguard/src/wireguardif.c: In function 'wireguardif_init':
.piolibdeps/esp07/esp_wireguard/src/wireguardif.c:961:5: error: 'err' is used uninitialized in this function [-Werror=uninitialized]
  if (err != ESP_OK) {
     ^
```

so I moved its initialization at the beginning of `wireguardif_init` function. And finally, just for future debugging, I changed some log messages.

Could you please test the branch `fix/eth` from my `esp_wireguard`? To be sure not to have mess up something.

```yaml
esphome:
# ...
  libraries:
    - esp_wireguard=https://github.com/droscy/esp_wireguard#fix/eth
```

If it is ok for you I'll merge it.

> With the patch, better messages are shown to indicate loss of connectivity:

Great, I will consider this also a test for the other issue ;-)

--- a/src/wireguardif.c
+++ b/src/wireguardif.c
@@ -939,16 +939,25 @@ err_t wireguardif_init(struct netif *netif) {
 
 	struct netif* underlying_netif = NULL;
 	char lwip_netif_name[8] = {0,};
+	// list of interfaces to try to bind wireguard to:
+	const char* ifkeys[2] = {"WIFI_STA_DEF", "ETH_DEF"};
+	
+	// ifkey will contain the selected interface key
+	const char* ifkey = NULL;
 
-	err = esp_netif_get_netif_impl_name(esp_netif_get_handle_from_ifkey("WIFI_STA_DEF"), lwip_netif_name);
+	ESP_LOGD(TAG, "Looking for available network interface");
+	for (int i = 0, err = ESP_FAIL; i < sizeof(ifkeys) / sizeof(char *) && err != ESP_OK; i++) {
+		ifkey = ifkeys[i];
+		err = esp_netif_get_netif_impl_name(esp_netif_get_handle_from_ifkey(ifkey), lwip_netif_name);
+	}
 	if (err != ESP_OK) {
-		ESP_LOGE(TAG, "esp_netif_get_netif_impl_name: %s", esp_err_to_name(err));
+		ESP_LOGE(TAG, "could not find a network interface. esp_netif_get_netif_impl_name: %s", esp_err_to_name(err));
 		result = ERR_IF;
 		goto fail;
 	}
 	underlying_netif = netif_find(lwip_netif_name);
 	if (underlying_netif == NULL) {
-		ESP_LOGE(TAG, "netif_find: cannot find WIFI_STA_DEF");
+		ESP_LOGE(TAG, "netif_find: cannot find %s", ifkey);
 		result = ERR_IF;
 		goto fail;
 	}

